src/js/core/methods/helpers/ethereumTypeDataConversions.js

```diff
@@ -0,0 +1,178 @@
+/* @flow */
+
+import BigNumber from 'bignumber.js';
+import type { EthereumSignTypedDataTypes } from '../../../types/networks/ethereum';
+import type { EthereumFieldType } from '../../../types/trezor/protobuf';
+import { Enum_EthereumDataType } from '../../../types/trezor/protobuf';
+import { ERRORS } from '../../../constants';
+import { stripHexPrefix } from '../../../utils/formatUtils';
+
+// Copied from https://github.com/ethers-io/ethers.js/blob/v5.5.2/packages/abi/src.ts/fragments.ts#L249
+const paramTypeArray = new RegExp(/^(.*)\[([0-9]*)\]$/);
+const paramTypeBytes = new RegExp(/^bytes([0-9]*)$/);
+const paramTypeNumber = new RegExp(/^(u?int)([0-9]*)$/);
+
+/**
+ * Parse the given EIP-712 array type into its entries, and its length (if not dynamic)
+ * E.g. `uint16[32]` will return `{entryTypeName: 'uint16', arraySize: 32}`.
+ */
+export function parseArrayType(arrayTypeName: string): {
+    entryTypeName: string,
+    arraySize: number | null,
+} {
+    const arrayMatch = paramTypeArray.exec(arrayTypeName);
+    if (arrayMatch === null) {
+        throw ERRORS.TypedError(
+            'Method_InvalidParameter',
+            `typename ${arrayTypeName} could not be parsed as an EIP-712 array`,
+        );
+    }
+    const [_, entryTypeName, arraySize] = arrayMatch;
+    return {
+        entryTypeName,
+        arraySize: parseInt(arraySize, 10) || null,
+    };
+}
+
+/**
+ * Converts a number to a two's compliment representation.
+ *
+ * E.g. -128 would be 0x80 in two's compliment, while 127 would be 0x7F.
+ *
+ * BigNumber.js has no built-in function, unlike https://www.npmjs.com/package/bn.js
+ */
+function twosCompliment(number: BigNumber, bytes: number): BigNumber {
```

Comment:
typo: compl**e**ment
```suggestion
function twosComplement(number: BigNumber, bytes: number): BigNumber {
```

Reply:
Good catch! Fixed in c2e9887de3b6c9e51296aad2701e3cf21f35d669